pandas/core/frame.py

```diff
@@ -5501,13 +5501,40 @@ def _replace_columnwise(
     @doc(NDFrame.shift, klass=_shared_doc_kwargs["klass"])
     def shift(
         self,
-        periods: int = 1,
+        periods: int | Iterable[int] = 1,
         freq: Frequency | None = None,
         axis: Axis = 0,
         fill_value: Hashable = lib.no_default,
+        suffix: str | None = None,
     ) -> DataFrame:
         axis = self._get_axis_number(axis)
 
+        if is_list_like(periods):
```

Comment:
This looks to me like something that could be factored out to a helper but I didn't see that pattern much in the rest of the code base, so I kept it here.

Reply:
There is a lot of logic below the is_list_like block; is this not necessary for the list-like case?

Reply:
Not that I can think of - remember that in the list-like case, we're calling the method _again_ but this time with an int for periods, and that time the rest of the logic is used.

Reply:
On L5531 you're calling super()'s method for the list-like case. That is not the same as calling self's, right? L5603 is also calling super()'s method, but with a lot of extra logic. And there are various code paths that never call super()'s method.

Reply:
I see - the logic below is for axis=1